docs/development/developer.rst

```diff
@@ -4,69 +4,108 @@ Developer Guide
+- ``policies``: modules defining policies and methods to manipulate them (e.g. serialization).
 
-- rewards
+- ``rewards``: modules to build, serialize and preprocess neural network based reward functions.
 
-- scripts
+- ``scripts``: command-line scripts for running experiments through sacred.
 
-- util
+- ``util``: provides utility functions like logging, configurations, etc.
 
 
 Algorithms
 ----------
 
-The base algorithm class implements the following two classes:
+The ``imitation.algorithms.base`` module defines the following two classes:
 
 - ``BaseImitationAlgorithm``: Base class for all imitation algorithms. 
 
-- ``DemonstrationAlgorithm``: Base class for all demonstration based algorithms like BC, IRL, etc. This class is inherited from ``BaseImitationAlgorithm``.
-    Demonstration algorithms offers following methods and properties:
-    - ``policy`` property that returns a policy imitating the demonstration data.
+- | ``DemonstrationAlgorithm``: Base class for all demonstration based algorithms like BC, IRL, etc. This class subclasses ``BaseImitationAlgorithm``. 
+  | Demonstration algorithms offers following methods and properties:
 
-    - ``set_demonstrations()`` method that sets the demonstrations data for learning.
+  - ``policy`` property that returns a policy imitating the demonstration data.
 
-All of the algorithms provide the ``train()`` method for training an agent using the algorithm.
+  - ``set_demonstrations()`` method that sets the demonstrations data for learning.
 
-All the available algorithms are present in algorithms/ with each algorithm in a distinct file. 
-Adversarial algorithms like AIRL and GAIL are present in algorithms/adversarial.
+All of the algorithms provide the ``train()`` method for training an agent and/or a reward network.
+
+All the available algorithms are present in ``algorithms/`` with each algorithm in a distinct file. 
+Adversarial algorithms like AIRL and GAIL are present in ``algorithms/adversarial``.
 
 
 Data
 ----
 .. automodule:: imitation.data
     :noindex:
 
-Envs
-----
-.. automodule:: imitation.envs
-    :noindex:
+    ``data.wrapper.BufferingWrapper``: Wraps a ``VecEnv`` to save the trajectories from all the environments
+    in a buffer.
 
-Policies
---------
-.. automodule:: imitation.policies
-    :noindex:
+    ``data.wrapper.RolloutInfoWrapper``: Wraps a ``gym.Env`` to log the original observations and rewards recieved from 
+    the environment. The observations and rewards of the entire episode is logged in the ``info`` dictionary with the
+    key "rollout" of the step where the episode ends. This wrapper is useful to save rollout trajectories, especially
+    in cases where you want to bypass the reward and/or observation overrides from other wrappers. 
+    See ``data.rollout.unwrap_traj`` for details and ``scripts/train_rl.py`` for example usecase.
+
+    ``data.rollout.rollout``: Generates rollout by taking in any policy as input along with the environment. 
+
+.. Envs
```

Comment:
Should probably delete this rather than leave it commented out? I'm fine not describing the environments in detail as https://github.com/HumanCompatibleAI/imitation/pull/541 in the process of moving them to another repo anyway.